src/wildcard-service/init.lua

```diff
@@ -6,4 +11,67 @@ local mt = {
   __index = _M
 }
 
+function _M.new(options)
+  local opts = options or {}
+  local backend = cache_backend.new(opts.client or resty_backend)
+
+  local http_client = http_ng.new({
+    backend = backend,
+    options = { ssl = { verify = resty_env.enabled('OPENSSL_VERIFY') or false } }
+  })
+
+  local api_host = opts.api_host or getenv('API_HOST') or 'system-provider:3000'
+  local access_token = opts.access_token or getenv('ACCESS_TOKEN')
+
+  return setmetatable({
+    options = opts,
+    http_client = http_client,
+    api_host = api_host,
+    access_token = access_token
+  }, mt)
+end
+
+function _M:get_apicast_servers()
+  local arg_host = _M.arg_host()
+  local query = ngx.encode_args({ host = arg_host, access_token = self.access_token })
+
+  if not arg_host then
+    ngx.log(ngx.WARN, 'missing host')
+    return ngx.exit(404)
+  end
+
+  for _, env in ipairs({'production', 'sandbox'}) do
```

Comment:
Just a nitpick, but LuaJIT likes more the `for i=1, #arr do ` loops than ipairs as they can be JITed. Also this allocates new table for each request, but the table can be property of the module.